### DISCUSSION TITLE
How can I change the information that needs to be input at runtime from image path to image？

### DISCUSSION THREAD

**Discussion by YourHotPotato** [ultralytics/ultralytics#14216] [Q&A]
model = YOLO("path/to/best.pt")  
results = model("https://ultralytics.com/images/bus.jpg")  

When using YOLOV8 for prediction, I need to input the path of the image, but in actual use, I want to process some intermediate process images, but the way of saving the intermediate process images first and then reading the path is rather cumbersome. May I ask how can I modify the YOLOV8 code to change the input from the image path to any image I want?
Thanks！

**Comment by pderrenger** [comment]
@YourHotPotato hello! 👋

You can directly pass image data to the YOLOv8 model without saving it to disk first. YOLOv8 supports various input types, including NumPy arrays, PIL images, and OpenCV images. Here's how you can modify your code to use an image directly:

### Using a NumPy Array
If you have the image as a NumPy array, you can pass it directly to the model:
```python
import numpy as np
from ultralytics import YOLO

# Load your model
model = YOLO("path/to/best.pt")

# Example: Create a dummy image (replace this with your actual image data)
image = np.random.randint(0, 255, (640, 640, 3), dtype=np.uint8)

# Run inference
results = model(image)

# Process results
results.show()  # Display the results
```

### Using a PIL Image
If you have the image as a PIL image, you can also pass it directly:
```python
from PIL import Image
from ultralytics import YOLO

# Load your model
model = YOLO("path/to/best.pt")

# Example: Open an image using PIL (replace this with your actual image data)
image = Image.open("path/to/your/image.jpg")

# Run inference
results = model(image)

# Process results
results.show()  # Display the results
```

### Using an OpenCV Image
If you are working with OpenCV, you can pass the image in the same way:
```python
import cv2
from ultralytics import YOLO

# Load your model
model = YOLO("path/to/best.pt")

# Example: Read an image using OpenCV (replace this with your actual image data)
image = cv2.imread("path/to/your/image.jpg")

# Run inference
results = model(image)

# Process results
results.show()  # Display the results
```

These methods allow you to bypass saving intermediate images to disk, making your workflow more efficient. For more details on supported input types, you can refer to the [Ultralytics documentation](https://docs.ultralytics.com/modes/predict/).

Feel free to reach out if you have any more questions! 😊

**Reply by YourHotPotato** [reply]
Thanks for your answer

**Reply by pderrenger** [reply]
@YourHotPotato you're welcome! 😊 If you have any more questions or run into any other issues, feel free to ask. We're here to help!

If you encounter any bugs or need further assistance, providing a minimum reproducible example can greatly help us diagnose and resolve the issue more efficiently. You can find more information on how to create one here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Also, please ensure you're using the latest version of the Ultralytics packages, as updates often include important bug fixes and improvements.

Happy coding! 🚀
